### PR TITLE
Use Rust 2018 features

### PR DESCRIPTION
**Pull request by KiChjang**
See individual commit for details.

### CODE REVIEW

**Comment by KiChjang** [comment]
Feel free to remove the commits that you don't think would fit in your project.

File: benches/perf_test.rs [hunk 1]
```diff
@@ -1,22 +1,18 @@
...
-use rusty_razor::chase::chase::solve_all;
-use criterion::Criterion;
-use criterion::criterion_group;
-use criterion::criterion_main;
-use rusty_razor::formula::parser::parse_theory;
-use rusty_razor::chase::selector::Bootstrap;
-use rusty_razor::chase::selector::Fair;
-use std::fs;
-use std::io::Read;
+use criterion::{Criterion, criterion_group, criterion_main};
```
Comment (salmans) [inline benches/perf_test.rs]: Nice!

File: src/chase.rs [hunk 2]
```diff
@@ -270,4 +272,4 @@ mod test_chase {
         assert_eq!("<R(e#0, e#1, e#2)>", _R_().app3(_e_0(), _e_1(), _e_2()).to_string());
         assert_eq!("<e#0 = e#1>", _e_0().equals(_e_1()).to_string());
     }
-}
```
Comment (salmans) [inline src/chase.rs]: 👍

File: src/chase/impl.rs [hunk 3]
```diff
@@ -0,0 +1 @@
+pub mod basic;
```
Comment (salmans) [inline src/chase/impl.rs]: Ah I see what's going on here! Cool!

File: src/lib.rs [hunk 4]
```diff
@@ -1,6 +1,3 @@
-#[macro_use]
```
Comment (salmans) [inline src/lib.rs]: huh!

**Comment by salmans** [comment]
@KiChjang this looks great! Thanks a lot!

**Comment by salmans** [comment]
LGTM

**Comment by salmans** [comment]
LGTM

**Pull request merged**